text/0000-pack-buildpack-new-alternatives.md

```diff
@@ -0,0 +1,216 @@
+
+A full invocation to generate `bash` scaffolding prompts for the values [currently documented as flags](https://buildpacks.io/docs/buildpack-author-guide/create-buildpack/building-blocks-cnb/#using-the-pack-cli).
+
+```bash
+pack buildpack new
+Use the arrow keys to navigate: ↓ ↑ → ←
+? Choose a project template:
+  ▸ bash
+    go
+
+```
+
+A full session includes the terminal prompts that the project scaffolding tool asks of the end user:
+
+```bash
+pack buildpack new
+Use the arrow keys to navigate: ↓ ↑ → ←
+? Choose a project template:
+  ▸ bash
+    go
+✔ Enter a directory in which to scaffold the project: bash_buildpack
+Use the arrow keys to navigate: ↓ ↑ → ←
+? Choose the buildpack API version (use the default if you are unsure):
+  ▸ 0.7
+    0.8
+✔ Enter an identifier for the buildpack: example/bash
+✔ Enter the initial buildpack version: 0.0.1
+✔ Enter a stack this buildpack will support by default: io.buildpacks.samples.stacks.bionic
+
+Created project in bash_buildpack
+```
+
+Templates are provided as a file tree.  A root-level `prompts.toml` file contains prompts for the end-user.  For example, a partial file tree containing user prompts and Go skeleton code is:
+
+```bash
+.
+├── prompts.toml
+└── {{.ProjectDirectory}}
+    ├── buildpack.toml
+    ├── cmd
+    │   └── main.go
+    ├── go.mod
+    └── pkg
+        ├── build.go
+        └── detect.go
+```
+
+Where `prompts.toml` is of the form:
+
+```toml
+[[prompt]]
+name="ProjectDirectory"
+prompt="Enter a directory in which to scaffold the project"
+default="bash_buildpack"
+required=true
+
+[[prompt]]
+name="BuildpackApi"
+prompt="Choose the buildpack API version (use the default if you are unsure)"
+choices=["0.7", "0.8"]
+
+[[prompt]]
+name="ModuleName"
+prompt="Enter a valid Go module name for this buildpack"
+default="github.com/user/buildpack"
+
+...
+```
+
+And an example templated source file is:
+
+```golang
+package main
+
+import (
+	"github.com/buildpacks/libcnb"
+
+	bp "{{ .ModuleName }}/pkg"
+)
+
+func main() {
+	libcnb.Main(
+		bp.Detect{},
+		bp.Build{},
+	)
+}
+```
+
+Template variables introduced in `prompts.yaml` are required to have a name unique within the `prompts.yaml` file.  An optional default value may be provided.
+
+Answers to prompts can be provided in a toml file.  This supports programmatic use of `pack`:
+
+```
+pack buildpack create --overrides overrides.toml
+Created project in bash_buildpack
+```
+
+The format of `--overrides` parallels the `prompts.toml` format.  An example of which is:
```

Comment:
Not sure that I agree with (or understand) this statement. The format / structure of the `overrides.toml` appears to be a simple key/value to the prompt names where as `prompts.yaml` is an array of tables.

Reply:
I have changed this to be a sequence of `--override` CLI flags.  Switching it to CLI flags was suggested on Slack.